.github/actions/make_init/action.yml

```diff
@@ -0,0 +1,88 @@
+name: "Setup Virtual Env"
+
+inputs:
+  use_cached_venv:
+    description: 'Use Cached Virtual Env'
+    default: 'true'
+
+runs:
+  using: composite
+  steps:
+    - name: Restore pre-commit cache
+      id: cache-pre-commit
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pre-commit
+        key: v1-pre-commit-${{ env.pythonLocation }}-${{ hashFiles('**/.pre-commit-config.yaml') }}
+    - if: steps.cache-pre-commit.outputs.cache-hit != 'true'
+      name: Install pre-commit
+      run: |
+        pip install pre-commit
+        pre-commit install-hooks
+      shell: bash -ileo pipefail {0}
```

Comment:
Small question: What's the significance of the `{0}` here? I'm unfamiliar with this syntax/argument/whatever it is

Reply:
This is the best explanation I found for it:
```
You can set the shell value to a template string using command […options] {0} [..more_options]. 
GitHub interprets the first whitespace-delimited word of the string as the command, and inserts 
the file name for the temporary script at {0}.
```
Without this, you will get the following error:
```
Error: Invalid shell option. Shell must be a valid built-in (bash, sh, cmd, powershell, pwsh) or a format string containing '{0}'
```

Reply:
Ah, okay that explains it. It's GH Actions weirdness when I was thinking it was something bash-specific.